WinSonic/Pages/Details/PlaylistDetailPage.xaml.cs

```diff
@@ -0,0 +1,223 @@
+        private bool IsOwner;
+
+        public PlaylistDetailPage()
+        {
+            InitializeComponent();
+            SongGridTable.Columns = [
+                new Tuple<string, GridLength>("Track", new GridLength(80, GridUnitType.Pixel)),
+                new Tuple<string, GridLength>("Title", new GridLength(4, GridUnitType.Star)),
+                new Tuple<string, GridLength>("Artist", new GridLength(3, GridUnitType.Star)),
+                new Tuple<string, GridLength>("Album", new GridLength(3, GridUnitType.Star)),
+                new Tuple<string, GridLength>("Time", new GridLength(80, GridUnitType.Pixel))
+            ];
+        }
+
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            base.OnNavigatedTo(e);
+            if (e.Parameter is InfoWithPicture info)
+            {
+                DetailedObject = info;
+            }
+        }
+
+        private async void Page_Loaded(object sender, RoutedEventArgs e)
+        {
+            if (DetailedObject != null)
+            {
+                Playlist = new(await SubsonicApiHelper.GetPlaylist(DetailedObject.ApiObject.Server, DetailedObject.ApiObject.Id));
+                OnPropertyChanged(nameof(Playlist));
+                IsOwner = string.Equals(Playlist.Playlist.Owner, Playlist.Playlist.Server.Username, StringComparison.OrdinalIgnoreCase);
+                OnPropertyChanged(nameof(IsOwner));
+                int i = 1;
+                foreach (var song in Playlist.Songs)
+                {
+                    TimeSpan duration = TimeSpan.FromSeconds(song.Duration);
+                    Dictionary<string, string?> dic = new()
+                    {
+                        ["Track"] = string.Format("{0:D" + Playlist.Songs.Count.ToString().Length + "}", i),
+                        ["Title"] = song.Title,
+                        ["Artist"] = song.Artist,
+                        ["Album"] = song.Album,
+                        ["Time"] = string.Format("{0:D1}:{1:D2}", duration.Minutes, duration.Seconds),
+                    };
+                    SongGridTable.AddRow(dic);
+                    i++;
+                }
+                SongGridTable.ShowContent();
+            }
+        }
+
+        private void PlayButton_Click(object sender, RoutedEventArgs e)
+        {
+            PlayerPlaylist.Instance.ClearSongs();
+            AddToQueueButton_Click(sender, e);
+        }
+
+        private void AddToQueueButton_Click(object sender, RoutedEventArgs e)
+        {
+            foreach (var song in Playlist.Songs)
+            {
+                PlayerPlaylist.Instance.AddSong(song);
+            }
+        }
+
+        private void PlayNextButton_Click(object sender, RoutedEventArgs e)
+        {
+            for (int i = Playlist.Songs.Count - 1; i >= 0; i--)
+            {
+                PlayerPlaylist.Instance.AddSong(Playlist.Songs[i], (int)app.MediaPlaybackList.CurrentItemIndex + 1);
```

Comment:
We should create a separate function on the PlayerPlaylist class to add the next song.